src/System.Management.Automation/commands/utility/FormatAndOutput/out-console/OutConsole.cs

```diff
@@ -143,15 +143,29 @@ protected override void EndProcessing()
             }
 
             base.EndProcessing();
+        }
 
-            if (Transcript)
+        /// <summary>
+        /// Revert transcription state on Dispose
+        /// </summary>
+        protected override void InternalDispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            base.InternalDispose();
```

Comment:
Maybe wrap this with `try`/`finally`, so you have opportunity to clean up even if `base.InternalDispose()` throw exception.